src/map/lua/lua_baseentity.cpp

```diff
@@ -4725,6 +4725,29 @@ int8 CLuaBaseEntity::getShieldSize()
     return 0;
 }
 
+/************************************************************************
+ *  Function: getShieldDefense()
+ *  Purpose : Return the defense of the equipped shield
+ *  Example : player:getShieldDefense()
+ *  Notes   : Returns 0 if player does not have shield equipped
+ ************************************************************************/
+
+int16 CLuaBaseEntity::getShieldDefense()
+{
+    if (m_PBaseEntity->objtype != TYPE_PC)
+    {
+        ShowWarning("Entity is not a Player or Pet type (%s).", m_PBaseEntity->getName());
+        return 0;
+    }
+
+    if (m_PBaseEntity->objtype == TYPE_PC)
+    {
+        return static_cast<CCharEntity*>(m_PBaseEntity)->getShieldDefense();
+    }
+
+    return 0;
+}
```

Comment:
```c++
int16 CLuaBaseEntity::getShieldDefense()
{
    if (m_PBaseEntity->objtype == TYPE_PC)
    {
        return static_cast<CCharEntity*>(m_PBaseEntity)->getShieldDefense();
    }
    else
    {
        ShowWarning("Entity is not a Player or Pet type (%s).", m_PBaseEntity->getName());
    }

    return 0;
}
```

Reply:
Refactored.

Reply:
and same here: this is the lua base entity file and that is the charentity function

Reply:
Fixed.